<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>863</width>
    <height>451</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>true</bool>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>0</number>
    </property>
    <property name="margin">
     <number>0</number>
    </property>
    <item>
     <widget class="QTabWidget" name="eTabs">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Load Profile</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="margin">
         <number>0</number>
        </property>
        <item>
         <widget class="QTableWidget" name="confTable">
          <property name="rowCount">
           <number>100</number>
          </property>
          <property name="columnCount">
           <number>2</number>
          </property>
          <attribute name="horizontalHeaderDefaultSectionSize">
           <number>150</number>
          </attribute>
          <attribute name="horizontalHeaderMinimumSectionSize">
           <number>150</number>
          </attribute>
          <attribute name="horizontalHeaderStretchLastSection">
           <bool>true</bool>
          </attribute>
          <attribute name="verticalHeaderVisible">
           <bool>false</bool>
          </attribute>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <row/>
          <column/>
          <column/>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButton_3">
          <property name="text">
           <string>Load Profile From File</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="selectTab">
       <attribute name="title">
        <string>Configure Experiment</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>2</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="6" colspan="2">
         <widget class="QFrame" name="frame_7">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="class" stdset="0">
           <string>PeripheralCollection</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_3">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>2</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <property name="horizontalSpacing">
            <number>3</number>
           </property>
           <property name="verticalSpacing">
            <number>5</number>
           </property>
           <item row="4" column="2">
            <widget class="QCommandLinkButton" name="commandLinkButton">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>30</height>
              </size>
             </property>
             <property name="text">
              <string>         Ready! Prepare the Experiment</string>
             </property>
            </widget>
           </item>
           <item row="0" column="0" rowspan="4">
            <widget class="QFrame" name="frame">
             <property name="toolTip">
              <string>Select the parameters that need to have a certain fixed value per dataset</string>
             </property>
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <property name="class" stdset="0">
              <string>ControlWidget</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout">
              <property name="spacing">
               <number>3</number>
              </property>
              <property name="leftMargin">
               <number>2</number>
              </property>
              <property name="rightMargin">
               <number>2</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="label_2">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>MANUAL CHANNELS</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item alignment="Qt::AlignTop">
               <widget class="QFrame" name="frame_2">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="frameShape">
                 <enum>QFrame::Panel</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Raised</enum>
                </property>
                <property name="class" stdset="0">
                 <string/>
                </property>
                <layout class="QVBoxLayout" name="staticLayout">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>5</number>
                 </property>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="0" column="1" rowspan="4">
            <widget class="QFrame" name="frame_10">
             <property name="toolTip">
              <string>select the parameters you need to vary in
order to generate a dataset</string>
             </property>
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <property name="class" stdset="0">
              <string>ControlWidget</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_2">
              <property name="spacing">
               <number>2</number>
              </property>
              <property name="leftMargin">
               <number>2</number>
              </property>
              <property name="rightMargin">
               <number>2</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QLabel" name="label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>SWEEP CHANNELS</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item alignment="Qt::AlignTop">
               <widget class="QFrame" name="frame_12">
                <property name="frameShape">
                 <enum>QFrame::Panel</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Raised</enum>
                </property>
                <layout class="QVBoxLayout" name="sweepLayout">
                 <property name="margin">
                  <number>2</number>
                 </property>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="0" column="2" rowspan="4">
            <widget class="QFrame" name="frame_11">
             <property name="toolTip">
              <string>Select the channels you need to monitor
in order to obtain the dataset</string>
             </property>
             <property name="frameShape">
              <enum>QFrame::NoFrame</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <property name="class" stdset="0">
              <string>ControlWidget</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_3">
              <property name="spacing">
               <number>2</number>
              </property>
              <property name="leftMargin">
               <number>2</number>
              </property>
              <property name="rightMargin">
               <number>2</number>
              </property>
              <property name="bottomMargin">
               <number>2</number>
              </property>
              <item>
               <widget class="QLabel" name="label_3">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>READBACK CHANNELS</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QFrame" name="frame_3">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="frameShape">
                 <enum>QFrame::Panel</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Raised</enum>
                </property>
                <layout class="QGridLayout" name="sampleReadbacks">
                 <property name="margin">
                  <number>2</number>
                 </property>
                 <property name="spacing">
                  <number>2</number>
                 </property>
                </layout>
               </widget>
              </item>
              <item alignment="Qt::AlignTop">
               <widget class="QFrame" name="frame_13">
                <property name="frameShape">
                 <enum>QFrame::Panel</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Raised</enum>
                </property>
                <layout class="QVBoxLayout" name="monitorLayout">
                 <property name="spacing">
                  <number>2</number>
                 </property>
                 <property name="margin">
                  <number>2</number>
                 </property>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QPushButton" name="pushButton_6">
             <property name="text">
              <string>Save Profile</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="runTab">
       <attribute name="title">
        <string>Make Measurements</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="margin">
         <number>2</number>
        </property>
        <property name="spacing">
         <number>2</number>
        </property>
        <item row="0" column="1">
         <widget class="QFrame" name="frame_8">
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="class" stdset="0">
           <string>PeripheralCollection</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <property name="margin">
            <number>2</number>
           </property>
           <property name="spacing">
            <number>2</number>
           </property>
           <item row="0" column="1">
            <widget class="QFrame" name="frame_6">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <property name="class" stdset="0">
              <string>PeripheralCollectionInner</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QPushButton" name="pushButton_5">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Evaluate All Rows</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QSpinBox" name="delayBox">
                <property name="suffix">
                 <string> mS settling time</string>
                </property>
                <property name="minimum">
                 <number>0</number>
                </property>
                <property name="maximum">
                 <number>1000</number>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QCommandLinkButton" name="commandLinkButton_2">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>30</height>
              </size>
             </property>
             <property name="font">
              <font>
               <italic>false</italic>
               <underline>false</underline>
               <strikeout>false</strikeout>
              </font>
             </property>
             <property name="text">
              <string>                Plot Selected Columns</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" colspan="3">
            <widget class="QFrame" name="frame_5">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_5">
              <property name="spacing">
               <number>2</number>
              </property>
              <property name="margin">
               <number>0</number>
              </property>
              <item>
               <widget class="QFrame" name="frame_4">
                <property name="frameShape">
                 <enum>QFrame::NoFrame</enum>
                </property>
                <property name="frameShadow">
                 <enum>QFrame::Plain</enum>
                </property>
                <layout class="QGridLayout" name="controlsLayout">
                 <property name="margin">
                  <number>2</number>
                 </property>
                 <property name="spacing">
                  <number>3</number>
                 </property>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QTableWidget" name="tbl">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
          <zorder>frame_6</zorder>
          <zorder>frame_5</zorder>
          <zorder>commandLinkButton_2</zorder>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="plotTab">
       <attribute name="title">
        <string>View plots</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <property name="margin">
         <number>3</number>
        </property>
        <item row="0" column="2">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QComboBox" name="plotList">
          <property name="minimumSize">
           <size>
            <width>200</width>
            <height>0</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>savePlots</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="4">
         <widget class="QFrame" name="frame_9">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="plot_area">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="margin">
            <number>0</number>
           </property>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>commandLinkButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>prepare()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>819</x>
     <y>430</y>
    </hint>
    <hint type="destinationlabel">
     <x>862</x>
     <y>448</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_5</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>runAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>356</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>388</x>
     <y>-19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>commandLinkButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>plotColumns()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>674</x>
     <y>432</y>
    </hint>
    <hint type="destinationlabel">
     <x>695</x>
     <y>462</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveData()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>852</x>
     <y>46</y>
    </hint>
    <hint type="destinationlabel">
     <x>886</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>deleteCurve()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>327</x>
     <y>-6</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_6</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>saveProfile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>221</x>
     <y>440</y>
    </hint>
    <hint type="destinationlabel">
     <x>227</x>
     <y>474</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>loadProfile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>457</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>584</x>
     <y>-11</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>prepare()</slot>
  <slot>clearColumn()</slot>
  <slot>runAll()</slot>
  <slot>plotColumns()</slot>
  <slot>addCustom()</slot>
  <slot>saveData()</slot>
  <slot>deleteCurve()</slot>
  <slot>saveProfile()</slot>
  <slot>loadProfile()</slot>
 </slots>
</ui>
